### PR TITLE
chore(demo-playwright): test for navigation

### PR DESCRIPTION
**Pull request by VidhiRambhia**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Addresses https://github.com/taiga-family/taiga-ui/issues/4912

## What is the new behaviour?

This PR adds a test for navigation

### CODE REVIEW

File: projects/demo-playwright/tests/addon-doc/navigation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import {tuiGoto} from '@demo-playwright/utils';
+import {expect, test} from '@playwright/test';
+
+test.describe(`Navigation`, () => {
+    test.use({viewport: {width: 1080, height: 522}});
```
[the remark this answers is not in the document]
Comment (VidhiRambhia) [reply projects/demo-playwright/tests/addon-doc/navigation.spec.ts]: Done

File: projects/demo-playwright/tests/addon-doc/navigation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    test(`getting started / [light mode]`, async ({page}) => {
+        await tuiGoto(page, `/getting-started`, {
+            hideHeader: false,
+            hideLanguageSwitcher: true,
+            hideVersionManager: true,
+        });
+        const example = page.locator(`tui-doc-navigation`);
```
[the remark this answers is not in the document]
Comment (VidhiRambhia) [reply projects/demo-playwright/tests/addon-doc/navigation.spec.ts]: Done

File: projects/demo-playwright/tests/addon-doc/navigation.spec.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    });
+
+    test(`getting started / [night mode]`, async ({page}) => {
+        await tuiGoto(page, `/getting-started`, {
+            hideHeader: false,
+            enableNightMode: true,
+            hideLanguageSwitcher: true,
+            hideVersionManager: true,
+        });
+        const example = page.locator(`tui-doc-navigation`);
```
[the remark this answers is not in the document]
Comment (VidhiRambhia) [reply projects/demo-playwright/tests/addon-doc/navigation.spec.ts]: Done

File: projects/demo-playwright/tests/addon-doc/navigation.spec.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        test(`scroll to \`tui-doc-code\``, async ({page}) => {
+            await tuiGoto(page, `/getting-started#icons`);
+            await expect(page.locator(`#icons`)).toBeVisible();
+            await expect(page.locator(`#icons`)).toBeInViewport();
+        });
+
+        test(`scroll after click on link with anchor`, async ({page}) => {
+            await tuiGoto(page, `/getting-started`);
+            const example = page.locator(`a[fragment="root"]`);
```
[the remark this answers is not in the document]
Comment (VidhiRambhia) [reply projects/demo-playwright/tests/addon-doc/navigation.spec.ts]: Done

File: projects/demo-playwright/tests/addon-doc/navigation.spec.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+            await expect(example).toBeVisible();
+            await example.click();
```
[the remark this answers is not in the document]
Comment (VidhiRambhia) [reply projects/demo-playwright/tests/addon-doc/navigation.spec.ts]: Makes sense

File: projects/demo-playwright/tests/addon-doc/navigation.spec.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            hideVersionManager: true,
+        });
+        const example = page.locator(`tui-doc-navigation`);
+
+        await example.isVisible();
+        await expect(example).toHaveScreenshot(`02-tui-doc-navigation-night-mode.png`);
+    });
+
+    test.describe(`anchor links navigation works`, () => {
+        test(`scroll to \`tui-doc-example\``, async ({page}) => {
```
[the remark this answers is not in the document]
Comment (VidhiRambhia) [reply projects/demo-playwright/tests/addon-doc/navigation.spec.ts]: Done

File: projects/demo-playwright/tests/addon-doc/navigation.spec.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            await tuiGoto(page, `/components/input#table`);
+            await expect(page.locator(`#table`)).toBeInViewport();
+        });
+
+        test(`scroll to \`tui-doc-code\``, async ({page}) => {
```
[the remark this answers is not in the document]
Comment (VidhiRambhia) [reply projects/demo-playwright/tests/addon-doc/navigation.spec.ts]: Done

**Review by splincode — APPROVED** [review]

**Pull request merged**